bokeh/charts/catheatmap.py

```diff
@@ -170,20 +130,28 @@ def get_source(self):
         and calculate the proper ranges.
         """
         self.source = ColumnDataSource(self.data)
-        self.xdr = FactorRange(factors=self.catsx)
-        self.ydr = FactorRange(factors=self.catsy)
+        self.x_range = FactorRange(factors=self.catsx)
+        self.y_range = FactorRange(factors=self.catsy)
 
     def draw(self):
         """Use the rect glyphs to display the categorical heatmap.
 
         Takes reference points from data loaded at the ColumnDataSurce.
         """
-        self.chart.make_rect(self.source, "catx", "caty", "width", "height",
-                             "color", "white", None)
-
-    def _show_teardown(self):
-        """Add hover tool to HetMap chart"""
-        self.chart.plot.add_tools(HoverTool(tooltips=[("value", "@rate")]))
+        glyph = Rect(
+            x="catx", y="caty",
+            width="width", height="height",
+            fill_color="color", fill_alpha=0.7,
+            line_color="white"
+        )
+        renderer =  GlyphRenderer(data_source=self.source, glyph=glyph)
+        # TODO: Legend??
+        # self._legends.append((self.groups[i], [renderer]))
+        yield renderer
+
+    # def _show_teardown(self):
+    #     """Add hover tool to HetMap chart"""
+    #     self.add_tools(HoverTool(tooltips=[("value", "@rate")]))
```

Comment:
I guess we want to keep the Hover... this is commented because is something to fix later?